### PR TITLE
Prevent leak of variables into deploy.php

### PR DESCRIPTION
**Pull request by mathbr**
| Q             | A
| ------------- | ---
| Bug fix?      | Yes
| New feature?  | No
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | #951

### CODE REVIEW

**Comment by glensc** [comment]
interesting solution. seems to solve variable overwrite and variable leak.

**Comment by elfet** [comment]
Nice, does it works on php5.6?

**Comment by glensc** [comment]
closures come with php 5.3, so it must work

however particular implementation i think lacks `use $deployFile` for the closure

**Comment by mathbr** [comment]
@glensc Good catch, I tried to prevent one variable too much from leaking. ;-)

**Comment by glensc** [comment]
LGTM now

File: bin/dep [hunk 1]
```diff
@@ -104,7 +104,10 @@ $deployer = new \Deployer\Deployer($console, $input, $output);
 
 // Require deploy.php file
 if (is_readable($deployFile)) {
-    require $deployFile;
+    // Prevent variable leak into deploy.php file
```
Comment (glensc) [inline bin/dep]: it's actually two way protection: leak into and leak from

**Pull request merged**